### DISCUSSION TITLE
Where should I put initial startup code for when the app begins?

### DISCUSSION THREAD

**Discussion by gfmoore** [dotnet/maui#7960] [Q&A]
New to this environment. Where should I put start up, initialisation code? For instance now that I can check Bluetooth permissions I think this code shouldn't be put in a particular page I call, but somewhere else. I'm thinking MauiProgram.cs, but perhaps I should create a class with some methods or an async task somewhere and call them from somehwere. 

I don't really know how this works though. I know I'm ignorant  of some basic C# stuff, but it's hard to find good places as to where to learn this stuff related to Maui and the MS documentation is almost impenetrable to me :(

**Comment by Symbai** [comment]
I would put it into the constructor of `App` (so in `App.xaml.cs` file) behind `InitializeComponent();` line. This ensures that everything important is loaded and its close before something like a page is displayed. But there are several ways and several locations, like your `MauiProgram.cs`. Depending what exactly you want to execute, multiple places can be correct or there is just one. If you are unsure, pick one, run it and see if it gives you the expected result.
